OpenRA.Mods.D2k/Traits/WormFood.cs

```diff
@@ -0,0 +1,53 @@
+#region Copyright & License Information
+/*
+* Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+* This file is part of OpenRA, which is free software. It is made
+* available to you under the terms of the GNU General Public License
+* as published by the Free Software Foundation. For more information,
+* see COPYING.
+*/
+#endregion
+
+using System.Collections.Generic;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
+{
+	[Desc("This actor attracts sandworms.")]
+	class WormFoodInfo : ITraitInfo
+	{
+		[Desc("List of terrains on which making noise can get you killed by a sandworm.")]
+		public static HashSet<string> DangerousTerrain = new HashSet<string> { "Sand", "Dune", "Spice", "SpiceBlobs" };
+
+		[Desc("How much does the sound subside with each unit of distance.")]
+		public static double Attenuation = 0.36;
```

Comment:
Two problems here:
- Making these static means that *all* actors will share the same value, which will be determined arbitrarily by the order that your unit definitions are parsed.
- Floating point numbers can't safely be used for gameplay code. You'll need to rewrite this as an integer - I suggest copying how the warhead Falloff works.